include/util/deallocating_vector.hpp

```diff
@@ -239,6 +239,19 @@ class DeallocatingVector
         bucket_list.emplace_back(new ElementT[ELEMENTS_PER_BLOCK]);
     }
 
+    DeallocatingVector(DeallocatingVector &&other)
+    {
+        bucket_list = std::move(other.bucket_list);
+        current_size = std::move(other.current_size);
+    }
+
+    DeallocatingVector &operator=(DeallocatingVector &&other)
+    {
+        bucket_list = std::move(other.bucket_list);
+        current_size = std::move(other.current_size);
+        return *this;
+    }
```

Comment:
I don't understand the move ctor and move assign op here. The dealloc. vec has a size_t and a vector member attribute. Why don't we initialize the size_t here? (The move ops are basically what swap is supposed to do).

https://github.com/Project-OSRM/osrm-backend/blob/86a31f1080c7fe458030b341a84deab3ec826b5f/include/util/deallocating_vector.hpp#L227-L228

Hm, `= default();`-ing both should have the same effect and be correct, no?